src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1355,19 +1355,24 @@ private void CheckDisallowedAttributeDependentType(TypeWithAnnotations typeArgum
         {
             typeArgument.VisitType(type: null, static (typeWithAnnotations, arg, _) =>
             {
-                var (topLevelType, errorLocation, diagnostics) = arg;
+                var (errorLocation, diagnostics) = arg;
                 var type = typeWithAnnotations.Type;
                 if (type.IsDynamic()
                     || (typeWithAnnotations.NullableAnnotation.IsAnnotated() && !type.IsValueType)
                     || type.IsNativeIntegerWrapperType
                     || (type.IsTupleType && !type.TupleElementNames.IsDefault))
                 {
-                    diagnostics.Add(ErrorCode.ERR_AttrDependentTypeNotAllowed, errorLocation, topLevelType.ToDisplayString(SymbolDisplayFormat.CSharpShortErrorMessageFormat));
+                    diagnostics.Add(ErrorCode.ERR_AttrDependentTypeNotAllowed, errorLocation, type.ToDisplayString(SymbolDisplayFormat.CSharpShortErrorMessageFormat));
+                    return true;
+                }
+                if (type.Kind == SymbolKind.TypeParameter)
```

Comment:
>if (type.Kind == SymbolKind.TypeParameter) [](http://example.com/codeflow?start=16&length=42)

Consider adding an empty line between two multi-line statements. This is going to improve readability of the code and probably will make the diff for the change easier to follow.

Reply:
Good idea. Fixed: b451f3daccb48971a89b4eb8cdcdeb2632167482